src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.cs

```diff
@@ -288,24 +288,47 @@ private static void ValidateBuffers<T>(IReadOnlyList<T> buffers)
         private static ValueTask<int> ScheduleSyncReadAtOffsetAsync(SafeFileHandle handle, Memory<byte> buffer,
             long fileOffset, CancellationToken cancellationToken, OSFileStreamStrategy? strategy)
         {
+            // We don't use Task anymore, which means that the work item
+            // can't be unqueued from the thread pool if the token gets canceled.
+            // But if it already is, we can avoid queueing the work item altogether.
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return ValueTask.FromCanceled<int>(cancellationToken);
+            }
+
             return handle.GetThreadPoolValueTaskSource().QueueRead(buffer, fileOffset, cancellationToken, strategy);
```

Comment:
The call sites appear to already be checking, e.g.
https://github.com/dotnet/runtime/blob/6ca8c9bc0c4a5fc1082c690b6768ab3be8761b11/src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.cs#L117-L122
Is this change making a meaningful difference?